dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResourceTest.java

```diff
@@ -0,0 +1,158 @@
+import com.dotcms.rest.api.v1.secret.view.HostView;
+import com.dotcms.rest.api.v1.secret.view.ServiceIntegrationHostView;
+import com.dotcms.rest.api.v1.secret.view.ServiceIntegrationView;
+import com.dotcms.security.secret.Param;
+import com.dotcms.security.secret.ServiceDescriptor;
+import com.dotcms.security.secret.Type;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Host;
+import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.common.collect.ImmutableMap;
+import com.liferay.portal.model.User;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Map.Entry;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response;
+import org.glassfish.jersey.media.multipart.BodyPart;
+import org.glassfish.jersey.media.multipart.FormDataMultiPart;
+import org.glassfish.jersey.media.multipart.file.StreamDataBodyPart;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ServiceIntegrationResourceTest extends IntegrationTestBase {
+
+    static final String ADMIN_DEFAULT_ID = "dotcms.org.1";
+    static final String ADMIN_DEFAULT_MAIL = "admin@dotcms.com";
+    static final String ADMIN_NAME = "User Admin";
+
+    private static ServiceIntegrationResource serviceIntegrationResource;
+
+    private static final ObjectMapper ymlMapper = new ObjectMapper(new YAMLFactory())
+            //.enable(SerializationFeature.INDENT_OUTPUT)
+            .setVisibility(PropertyAccessor.FIELD, Visibility.ANY)
+            .findAndRegisterModules()
+            ;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+
+        final User user = mock(User.class);
+        when(user.getUserId()).thenReturn(ADMIN_DEFAULT_ID);
+        when(user.getEmailAddress()).thenReturn(ADMIN_DEFAULT_MAIL);
+        when(user.getFullName()).thenReturn(ADMIN_NAME);
+        when(user.getLocale()).thenReturn(Locale.getDefault());
+        when(user.isBackendUser()).thenReturn(true);
+
+        final WebResource webResource = mock(WebResource.class);
+        final InitDataObject dataObject = mock(InitDataObject.class);
+        when(dataObject.getUser()).thenReturn(user);
+
+        when(webResource.init(any(WebResource.InitBuilder.class))).thenReturn(dataObject);
+
+        //ServiceIntegrationAPI serviceIntegrationAPI = new ServiceIntegrationAPIImpl();
+
+        ServiceIntegrationHelper serviceIntegrationHelper = new ServiceIntegrationHelper();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'serviceIntegrationHelper' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)